### ISSUE TITLE
Replace "osalbool" and "boolean" types with C99 bool

### ISSUE THREAD

**Issue by skliper** [nasa/cFE#350]
CCB intends to deprecate the OSAL-provided "boolean" and "osalbool" types in version 5.0.0, in favor of the C99-specified "bool" type.

CFE needs to be updated to stop using these types along with the related macro definitions:

    `osalbool` and `boolean` should be `bool`
    `TRUE` should be `true`
    `FALSE` should be `false`

**Comment by skliper** [comment]
_Imported from trac issue 319.  Created by jphickey on 2019-08-02T10:25:30, last modified: 2019-08-29T13:02:04_

**Comment by skliper** [comment]
_Trac comment by jphickey on 2019-08-03 09:06:27:_

Ready for review, commit [changeset:04d32e5] on branch `trac-319-use-c99-bool`

Note, this is a big changeset of over 1300 lines because it changes all use of TRUE/FALSE to be true/false instead.  The changes are almost all mechanical (search/replace) and do not change the logic.

A specific exception worth noting is that the `CFE_ES_RunLoop()` function is defined as returning a boolean true/false value in the API/doxygen documentation, but was prototyped as returning an `int32`.  An int32 return is easily confused with the CFE return codes (where CFE_SUCCESS is 0, basically the opposite).  To avoid possible error, this corrects the prototype to return a `bool` as it should.  Apps use this as a conditional in the `while` loop of their main routine.

**Comment by skliper** [comment]
_Trac comment by jphickey on 2019-08-22 10:12:50:_

Note this is a dependency of [cfs_osal:228] and must be merged before that one (or at the same cycle).

**Comment by skliper** [comment]
_Trac comment by jhageman on 2019-08-22 10:36:16:_

Approved for 'fast-track' to IC branch for test.  CCB will code review prior to merge to master.

**Comment by skliper** [comment]
_Trac comment by jphickey on 2019-08-22 16:06:59:_

Merged into branch `ic-ccb-20190821`

**Comment by skliper** [comment]
_Trac comment by jphickey on 2019-08-29 10:48:13:_

Confirmed that `ic-ccb-20190821` merge is acceptable for development merge

**Comment by skliper** [comment]
_Trac comment by jphickey on 2019-08-29 10:53:33:_

Merged to development branch

**Comment by skliper** [comment]
_Trac comment by jhageman on 2019-08-29 13:02:04:_

CCB 8/28/2019 - Code review and approved for merge to development
